pkg/openwsn/patches/0011-openstack-04-TRAN-add-blocking-sock.patch

```diff
@@ -0,0 +1,549 @@
++#define _TIMEOUT_MAGIC     (0xF38A0B63U)
++#define _TIMEOUT_MSG_TYPE  (0x8474)
++#endif /* MODULE_ZTIMER */
++#endif /* MODULE_OPENWSN */
+ 
+-sock_udp_t* udp_socket_list;
++//=========================== variables =======================================
+ 
+-// =========================== variables =======================================
+-// =========================== prototypes ======================================
++static sock_udp_t *_udp_socket_list;
++
++//=========================== prototypes ======================================
+ 
+ static bool _sock_valid_af(uint8_t af);
+ 
+@@ -29,10 +47,14 @@ static void _sock_get_local_addr(open_addr_t* local);
+ 
+ static void _sock_transmit_internal(void);
+ 
++#ifdef MODULE_ZTIMER
++static void _timeout_cb(void *arg);
++#endif
++
+ // ============================= public ========================================
+ 
+ void sock_udp_init(void) {
+-    udp_socket_list = NULL;
++    _udp_socket_list = NULL;
+ }
+ 
+ int sock_udp_create(sock_udp_t* sock, const sock_udp_ep_t* local, const sock_udp_ep_t* remote, uint16_t flags) {
+@@ -42,10 +64,14 @@ int sock_udp_create(sock_udp_t* sock, const sock_udp_ep_t* local, const sock_udp
+         return -EINVAL;
+     }
+ 
++#ifdef MODULE_OPENWSN
++    mbox_init(&sock->mbox, sock->mbox_queue, OPENWSN_SOCK_MBOX_SIZE);
++#endif
++
+     memset(&sock->gen_sock.local, 0, sizeof(sock_udp_ep_t));
+ 
+     if (local != NULL) {
+-        current = udp_socket_list;
++        current = _udp_socket_list;
+ 
+         while (current != NULL) {
+             if (current->gen_sock.local.port == local->port) {
+@@ -73,10 +99,13 @@ int sock_udp_create(sock_udp_t* sock, const sock_udp_ep_t* local, const sock_udp
+     }
+ 
+     sock->gen_sock.flags = flags;
++
++#ifdef SOCK_HAS_ASYNC
+     sock->async_cb = NULL;
++#endif
+ 
+-    sock->next = udp_socket_list;
+-    udp_socket_list = sock;
++    sock->next = _udp_socket_list;
++    _udp_socket_list = sock;
+ 
+     return 0;
+ }
+@@ -144,21 +173,21 @@ int sock_udp_send(sock_udp_t* sock, const void* data, size_t len, const sock_udp
+ 
+     memcpy(pkt->payload, data, len);
+ 
+-    scheduler_push_task(_sock_transmit_internal, TASKPRIO_UDP);
+-
+     pkt->l4_payload = pkt->payload;
+     pkt->l4_length = pkt->length;
+ 
++    scheduler_push_task(_sock_transmit_internal, TASKPRIO_UDP);
++
+     return len;
+ }
+ 
+ void sock_udp_close(sock_udp_t* sock) {
+-    sock_udp_t* temp = udp_socket_list;
+-    sock_udp_t* prev = udp_socket_list;
++    sock_udp_t* temp = _udp_socket_list;
++    sock_udp_t* prev = _udp_socket_list;
+ 
+     /* check if head is the socket to be closed */
+     if (temp != NULL && temp == sock) {
+-        udp_socket_list = temp->next;
++        _udp_socket_list = temp->next;
+ 
+         return;
+     }
+@@ -201,28 +230,69 @@ int sock_udp_get_remote(sock_udp_t* sock, sock_udp_ep_t* ep) {
+ int sock_udp_recv(sock_udp_t* sock, void* data, size_t max_len, uint32_t timeout, sock_udp_ep_t* remote) {
```

Comment:
The signature changed here to:

```C
ssize_t sock_udp_recv_aux(sock_udp_t *sock, void *data, size_t max_len,
                          uint32_t timeout, sock_udp_ep_t *remote,
                          sock_udp_aux_rx_t *aux);
```

The `aux` parameter can be left unused. (And IMO it makes sense to add this feature as a follow up.)

The original `sock_aux_recv()` function remains in place for users as `static inline` wrapper on top of `sock_aux_recv()`.